### PR TITLE
HW2 Add additional configuration options.

### PR DESCRIPTION
**Pull request by vm137**
Danila, I changed configuration due to your comments.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "extends": "airbnb"
+}
```
Comment (loginov-rocks) [inline .eslintrc:59]: Please add "parser": "babel-eslint" and it'll be fixed

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,3 @@
-.idea/
-
 node_modules/
+
```
Comment (loginov-rocks) [inline .gitignore:3]: Consider adding .idea/workspace.xml file to gitignore, it contains user-specific settings only.

File: package.json [hunk 3]
```diff
@@ -1,19 +1,19 @@
 {
   "name": "movies-db",
   "version": "1.0.0",
-  "description": "react-redux movies database",
+  "description": "react movie database",
   "main": "index.js",
   "scripts": {
     "start": "webpack-dev-server",
```
Comment (loginov-rocks) [inline package.json]: Consider adding "lint": "eslint src" command to run lint from console like so: "npm run lint"

File: package.json [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
-    "build:dev": "webpack --env dev",
-    "build:prod": "webpack --env prod",
+    "build:dev": "webpack",
```
Comment (loginov-rocks) [inline package.json]: If mode is not specified it is "production" by default. More adjustments for development process https://webpack.js.org/guides/development/

File: public/index.html [hunk 5]
```diff
@@ -7,5 +7,8 @@
 <body>
     <div id='app'></div>
     <script src='/bundle.js'></script>
+
+    <!--for local version-->
+    <!--<script src='../dist/bundle.js'></script>-->
```
Comment (loginov-rocks) [inline public/index.html]: You can use html-webpack-plugin that can attach produced bundles to the HTML depending on configuration https://github.com/jantimon/html-webpack-plugin

File: package.json (hunk 4 above)
Comment (loginov-rocks) [reply package.json]: I see you set "mode" inside webpack.config.js, but it still can be not straightforward to other developers :)

File: webpack.config.js [hunk 6]
```diff
@@ -1,31 +1,45 @@
-let path = require('path');
+const path = require('path');
 
-module.exports = env => {
-    console.log('env: ', env);
+const config = {
+  entry: path.resolve(__dirname, 'src/client/index'),
+  output: {
+    path: `${__dirname}/dist`,
```
Comment (loginov-rocks) [inline webpack.config.js]: Can be changed to path.resolve(__dirname, 'dist') for consistency, but also I'm not sure it's needed to specify full path.

**Review by loginov-rocks — CHANGES_REQUESTED** [review]

File: .eslintrc (hunk 1 above)
Comment (vm137) [reply .eslintrc:59]: I added this option, but I still have error for:  JSX not allowed in files with extension '.js'.
What can be missing?

File: public/index.html (hunk 5 above)
Comment (vm137) [reply public/index.html]: I hope I got you right.

File: .eslintrc (hunk 1 above)
Comment (vm137) [reply .eslintrc]: Additionally I've found some eslint.config on the web, it has fewer issues, but still some underlines. 
I wondering is it there really right eslint.config for React?

File: package.json (hunk 4 above)
Comment (vm137) [reply package.json]: Did I understand you correctly?

File: webpack.config.js [hunk 7]
```diff
@@ -1,31 +1,47 @@
...
+};
+
+module.exports = (env, argv) => {
+
+  if (argv.mode === 'production') {
+    config.mode = 'production';
+  }
+  if (argv.mode === 'development') {
+    config.mode = 'development';
+  }
```
Comment (vm137) [inline webpack.config.js]: @loginov-rocks is that what you meant with --mode options?

File: public/index.html (hunk 5 above)
Comment (vm137) [reply public/index.html]: ok. then I need to write template for div#app which is pretty much the same. 
so I will leave it as is.

File: package.json [hunk 8]
```diff
@@ -1,19 +1,22 @@
...
   "version": "1.0.0",
-  "description": "react-redux movies database",
+  "description": "react movie database",
   "main": "index.js",
   "scripts": {
-    "start": "webpack-dev-server",
-    "build:dev": "webpack --env dev",
-    "build:prod": "webpack --env prod",
+    "build:dev": "webpack-dev-server",
+    "build:prod": "webpack --mode=production",
```
Comment (vm137) [inline package.json:8]: @loginov-rocks, is it ok to leave this way for development version?

**Comment by vm137** [comment]
@loginov-rocks, I did changes for your last comments. 
the biggest question which is still left - with eslint configuration. I still have errors.

File: .eslintrc (hunk 1 above)
Comment (loginov-rocks) [reply .eslintrc:59]: This is the rule requirements, so the options are: A) change extensions for each file with JSX, B) configure ESLint, refer to https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-filename-extension.md#rule-options

File: package.json (hunk 8 above)
Comment (loginov-rocks) [reply package.json:8]: `npm start` is a common name for building development version and serving it on localhost, so I suggest to leave 'start' and 'build' options, the first one is webpack-dev-server, the latter is for "webpack --mode=production"

**Review by loginov-rocks — CHANGES_REQUESTED** [review]

File: webpack.config.js (hunk 7 above)
Comment (loginov-rocks) [reply webpack.config.js]: Webpack consumes mode argument by itself, so this is not needed.

**Pull request merged**